src/init.cpp

```diff
@@ -1568,6 +1574,12 @@ bool AppInitMain()
                 return InitError(strLoadError);
             }
         }
+
+        if (gArgs.GetBoolArg("-txindex", DEFAULT_TXINDEX)) {
+            auto txindex_db = MakeUnique<TxIndexDB>(nTxIndexCache, false, fReset);
+            g_txindex.reset(new TxIndex(std::move(txindex_db)));
```

Comment:
This is a lot of moves, since you already move TxIndex::TxIndex. Can you change to create the unique_ptr in the constructor call, e.g.:

```
g_txindex.reset(new TxIndex(MakeUnique<...>(....));
```

Or even better, use argument forwarding.

Reply:
I'd rather have them on separate lines because I think otherwise there is too much happening on one line (and it would get pretty long).

Regarding argument forwarding, even if `TxIndex` takes an rvalue-ref, the `std::move` is necessary.